### DISCUSSION TITLE
What is the default tab width for the templ formatter?

### DISCUSSION THREAD

**Discussion by RafaelZasas** [a-h/templ#610] [Q&A] [answered]
### Templ version is:  `templ v0.2.598`.

I don't actually know if templ is defining the tab width or if it is coming from somewhere in my neovim config (Which I recently changed).

This is an example of a templ component:

```html
templ Development(d models.Development, isAdmin bool) {
	<div class="container mx-auto p-4">
		<h1 class="text-3xl text-center py-4 font-bold text-primary">
			{ d.Name }
		</h1>
		<div class="shadow">
			@featuredImage(d)
			@thumbnailsGallery(d)
			<!-- Details Section -->
			@DevelopmentDetails(d, isAdmin)
			@floorPlan(d)
		</div>
	</div>
}
```

If I have more than 3 levels of nested elements, I have to horizontally scroll the page. Ideally I would prefer to have a max line width of 80, and if the tab width was halved, I doubt I would go over 80 chars.

I normally use prettier to format my html and don't think about this sort of thing.

Can someone please share some insight into the formatting rules and whether this is implemented by the templ formatter or somewhere else in my config.

**Comment by joerdav** [comment, marked as answer]
Tab width is not something that templ could define, it will lie with your editor.

**Comment by RafaelZasas** [comment]
@joerdav I am experiencing some weird behavior that makes me think otherwise.

I have quadruple checked on multiple different filetypes, including the go files within the server, and they all have 2 spaces as the tab width, as defined by my neovim config:

```lua
vim.opt.tabstop = 2
vim.opt.softtabstop = 2
vim.opt.shiftwidth = 2
vim.opt.expandtab = true
vim.opt.autoindent = true
```

I only get the strange 8 char tab width in templ files, such as this:

```html
templ showcase(developments []models.Development) {
	<div class="space-y-36 py-10">
		for _, d := range developments {
			<div
				class={
					"flex items-center justify-between",
					"md:even:flex-row-reverse md:odd:flex-row flex-col",
				}
			>
				<div
					class={
						"parallax md:w-[75%] w-full aspect-square rounded-lg bg-fixed",
						bgImage(utils.GetPrimaryImage(d.Images)),
					}
				></div>
				<div
					class="p-8 rounded-lg shadow-lg relative z-10
                    prose w-full md:max-w-[35%]"
				>
					<a
						href={ templ.SafeURL(fmt.Sprintf("/developments/%d", d.ID)) }
						class="text-2xl link font-bold"
					>
						{ d.Name }
					</a>
					<p>{ d.Description }</p>
				</div>
			</div>
		}
	</div>
}
```

For example, this is part of a html file that I have formatted via format on save:

```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, maximum-scale=5"
    />
```

It has 2 spaces as tab indents, as expected.

The only filetype I have an issue with is templ.

**Reply by joerdav** [reply]
Oh, I think I misunderstood, it's not that you don't like how wide the tabs are, it's that it isn't being indented with spaces?

This is inline with Golang, and a feature we adopted since templ is a superset of Go, not a decision made by templ, apologies for the confusion.

**Reply by RafaelZasas** [reply]
Well, that actually does explain why the files templ files look that way when I cat the files, but it is not exactly the issue I am facing.

I am seeing a difference in not only the html formatting in templ files, but the go too.

Take this function:
```go
func testFunc() string {
	str := "test"
	for range 10 {
		fmt.Println(str)
		str += " test"
	}
	return str
}
```

It looks like this in the `.go` file:
![image](https://github.com/a-h/templ/assets/42390827/e68afa5c-7b6c-4863-92b4-3d1e7588ec51)

But like this in the templ file:
![image](https://github.com/a-h/templ/assets/42390827/56a54c9d-1439-440b-951f-ee80ec0548f2)

I guess I am confused as to why `.go` files respects the 2 space tab width, but `.templ` files do not.
In the go file, the end of the `Println` closing parentheses is **column 20**, but in the templ file, the same character of the same function is in **column 32**

**Comment by RafaelZasas** [comment]
Also, if I run templ fmt from the terminal, and cat the file, I get the same issue, which is making me believe that it is in fact templ that is enforcing the tabs, and not my editor. I can set the tab width in neovim, which makes it look much better, but when I copy the code and paste (as above) I see that it is actually using tabs to format the file

**Comment by RafaelZasas** [comment]
@joerdav Sorry about this. Turns out it was the vim-sleuth plugin that was changing the tabstop to 8.
